Description: Disable gfxpayload=keep by default
 Setting gfxpayload=keep has been known to cause efifb to be inappropriately
 enabled.  In any case, with the current Linux kernel the result of this
 option is that early kernelspace will be unable to print anything to the
 console, so (for example) if boot fails and you end up dumped to an
 initramfs prompt, you won't be able to see anything on the screen.  As such
 it shouldn't be enabled by default in Debian, no matter what kernel options
 are enabled.
 .
 gfxpayload=keep is a good idea but rather ahead of its time ...
Author: Colin Watson <cjwatson@debian.org>
Bug-Debian: http://bugs.debian.org/567245
Forwarded: no
Last-Update: 2010-03-09

Index: b/util/grub.d/10_linux.in
===================================================================
--- a/util/grub.d/10_linux.in
+++ b/util/grub.d/10_linux.in
@@ -77,17 +77,7 @@
 
   # Use ELILO's generic "efifb" when it's known to be available.
   # FIXME: We need an interface to select vesafb in case efifb can't be used.
-  if [ "x$GRUB_GFXPAYLOAD_LINUX" = x ]; then
-      cat << EOF
-	load_video
-EOF
-      if grep -qx "CONFIG_FB_EFI=y" "${config}" 2> /dev/null \
-	  && grep -qx "CONFIG_VT_HW_CONSOLE_BINDING=y" "${config}" 2> /dev/null; then
-	  cat << EOF
-	set gfxpayload=keep
-EOF
-      fi
-  else
+  if [ "x$GRUB_GFXPAYLOAD_LINUX" != x ]; then
       if [ "x$GRUB_GFXPAYLOAD_LINUX" != xtext ]; then
 	  cat << EOF
 	load_video
